test/Microsoft.Extensions.Localization.Tests/ResourceManagerStringLocalizerTest.cs

```diff
@@ -59,10 +59,33 @@ public void EnumeratorCacheIsScopedByAssembly()
             Assert.Equal(expectedCallCount, resourceAssembly2.GetManifestResourceStreamCallCount);
         }
 
+        [Fact]
+        public void ResourceManagerStringLocalizer_GetAllStrings_MissingResourceDoesntThrow()
```

Comment:
If we do catch the excetions like I suggested below then there will be no throw case (though I should add a case for not following parents.)